### PR TITLE
Reply To feature API changes

### PR DESCRIPTION
**Pull request by aforge**
* `replyPreview` function to prepare and make reply quotes out of the originals.
* `QQ` style for formatting quotes.
* `IC` style for icons to be shown in reply quotes for attachments and other entities (e.g. forms).

### CODE REVIEW

File: src/drafty.js [hunk 1]
```diff
@@ -1582,6 +1763,136 @@ Drafty.getContentType = function() {
   return 'text/x-drafty';
 }
 
+// Removes reply quotes and any text/formatting associated with these.
+// Returns a Drafty document free of reply quotes.
+function stripQuotes(original) {
```
Comment (or-else) [inline src/drafty.js:1768]: I think this is excessively complex and partially replicates existing code. The idea here is to convert Drafty to a tree of spans, then to prune branches. `toTree` already converts span array to a tree.

If you want, I can take a shot at this.

File: src/drafty.js [hunk 2]
```diff
@@ -1653,7 +1967,9 @@ Drafty.preview = function(original, length) {
           style.tp = '' + st.tp;
         } else if (Array.isArray(ent) && ent.length > st.key && typeof ent_refs[st.key] == 'number') {
           style.key = ent_refs[st.key];
-          preview.ent[style.key] = copyLight(ent[st.key]);
+          const src = ent[st.key];
```
Comment (or-else) [inline src/drafty.js:1970]: This is not guaranteed to be defined.

File: src/drafty.js [hunk 3]
```diff
@@ -1653,7 +1967,9 @@ Drafty.preview = function(original, length) {
...
+          let copiedEnt = copyEnt(src, true);
```
Comment (or-else) [inline src/drafty.js:1971]: And if `ent[st.key]` does not exist this crashes.

File: src/drafty.js [hunk 4]
```diff
@@ -1688,3 +2009,93 @@ function copyLight(ent) {
...
+ * to be suitable for reply quote previews.
+ * @memberof Drafty
+ * @static
+ *
+ * @param {Drafty} original - Drafty object to shorten.
+ * @param {number} length - length in characters to shorten to.
+ * @param {StyleTransform} transform - style transformation callback. 
+ * @returns new shortened Drafty object leaving the original intact.
+ */
+Drafty.replyPreview = function(original, length, transform) {
```
Comment (or-else) [inline src/drafty.js:2039]: Is it possible to generalize `Drafty.preview` to make it suitable for both? There is a lot of code duplication between `Drafty.preview` and `Drafty.replyPreview`.

File: src/drafty.js [hunk 5]
```diff
@@ -1224,9 +1399,61 @@ Drafty.UNSAFE_toHTML = function(content) {
   return txt;
 }
 
+// Denormalizes formatting and entity data into a simple span array.
+function prepareSpans(fmt, ent) {
```
Comment (or-else) [inline src/drafty.js:1403]: There is `spannify` and `prepareSpans` while spans they produce are completely different. Maybe generate the same format of spans or somehow un-confuse the method names?

File: src/drafty.js (hunk 1 above)
Comment (or-else) [reply src/drafty.js:1720]: Sure.

**Review by aforge — COMMENTED** [review]
Refactored `Drafty.preview` (merged `preview` and `replyPreview` into one) and `stripQuotes` (turn original into a tree and prune `QQ` branches) as suggested.

**Pull request merged**